lighthouse-core/lib/page-functions.js

```diff
@@ -169,6 +169,26 @@ function getOuterHTMLSnippet(element, ignoreAttrs = [], snippetCharacterLimit =
   }
 }
 
+/**
+ * Get the maximum size of a texture the GPU can handle
+ * @see https://bugs.chromium.org/p/chromium/issues/detail?id=770769#c13
+ */
+/* istanbul ignore next */
+function getMaxTextureSize() {
+  try {
+    let canvas = document.createElement('canvas');
+    let gl = canvas.getContext('webgl');
+    const maxTextureSize = gl.getParameter(gl.MAX_TEXTURE_SIZE);
+    canvas = gl = undefined; // Cleanup for GC
```

Comment:
I thought there might have been a more explicit "destroy" method for a context, but there isn't. As long as the element is not on the DOM, we're good. Can remove the "cleanup" line.

sorry for confusion

Reply:
maybe i'm overly cautious but memory mgmt with webgl feels like it has some gotchas.. as such, i'm prone to keep the explicit cleanup and not just wait for auto-GC

Reply:
@mathiasbynens we're just two developers who don't know how GC works. wdyt? :)

Reply:
Neither `canvas` nor the `gl` context escapes from this block, so it _should_ in theory be fine even without the explicit cleanup. (In case anyone‘s interested, I wrote a high-level explanation of [escape analysis](https://v8.dev/blog/disabling-escape-analysis) when I learned about it.)

That said, I’m not familiar with Chromium’s WebGL context lifecycle, and if perhaps there are weird side effects to creating such contexts that I don’t know about.